packages/components/src/helperMessage/src/HelperMessage.tsx

```diff
@@ -0,0 +1,66 @@
+import { type StyledComponentProps, useStyledSystem } from "@hopper-ui/styled-system";
+import { forwardRef, type ForwardedRef, type CSSProperties } from "react";
+import clsx from "clsx";
+import { HelperMessageContext } from "./HelperMessageContext.ts";
+import { useContextProps } from "react-aria-components";
+import { InfoIcon } from "@hopper-ui/icons";
+import { type TextProps, Text } from "../../Text/index.ts";
+import { cssModule } from "../../utils/src/css-module.ts";
+
+import styles from "./HelperMessage.module.css";
+
+export const GlobalHelperMessageCssSelector = "hop-HelperMessage";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultHelperMessageSlot = "description";
```

Comment:
This use case is a little different than the other. It's the first time that the slot here is an actual slot to fill.

The use case the default slot tried to solve

Let's say you have a button
```
<Button>
    <InfoIcon />
    <Text>Button content</Text>
    <InfoIcon slot="end-icon" />
```

If i wanted to use the SlotProvider to pass a className to the start and end icon, without having to specify "slot=start-icon" to the first InfoIcon
```
<Provider value={[
    [ButtonContext]: {
          // i can write props here that are passed to both slotted and unslotted buttons
          data-randomProps: 1;
         slots: {
             "end-icon": {
                  //i can pass properties to the end-icon here               
             },
       }]
]}
```
but there was no way of targetting the "unslotted" or "defaultSlotted" element. So i added the "button" slot by default to every buttons, so i could target them 

```
<Provider value={[
    [ButtonContext]: {
          // i can write props here that are passed to both slotted and unslotted buttons
          data-randomProps: 1;
         slots: {
             "end-icon": {
                  //i can pass properties to the end-icon here               
             },
             "button": { 
                  // i can pass things to the start button now
             }
       }]
]}
```

In the next release, i'll now be able to remove the "button" slot by default and do: 


```
import {DEFAULT_SLOT} from "react-aria-components"

<Provider value={[
    [ButtonContext]: {
          // i can write props here that are passed to both slotted and unslotted buttons
          data-randomProps: 1;
         slots: {
             "end-icon": {
                  //i can pass properties to the end-icon here               
             },
             [DEFAULT_SLOT]: { 
                  // i can pass things to the start button now
             }
       }]
]}
```

In your case, since you will never have 2 helper message slotted in different slots, you don't need a default slot for the HelperMessage. But the text message inside the component needs it to be positionned properly. 
```
    [props, ref] = useContextProps(props, ref, HelperMessageContext);
    const { stylingProps, ...ownProps } = useStyledSystem(props);
    const { className, children, showIcon, style, slot = "description", , ...otherProps } = ownProps;
```

Reply:
ok thank you for clarifying. I think I was a bit confused about the whole slot situation and why you were doing what you did.